src/wp-includes/webfonts-api/class-wp-webfonts-provider-registry.php

```diff
@@ -0,0 +1,108 @@
+<?php
+/**
+ * Webfonts API: Provider Registry
+ *
+ * @package WordPress
+ * @subpackage Webfonts
+ * @since 5.9.0
+ */
+
+/**
+ * Provider Registry.
+ *
+ * This registry exists to handle all providers.
+ *
+ * It handles the following within the API:
+ *  - loads the bundled provider files into memory;
+ *  - registers each provider with the API by:
+ *       1. creating an instance (object);
+ *       2. storing it in-memory (by its unique provider ID) for use with the API;
+ *
+ * @since 5.9.0
+ */
+class WP_Webfonts_Provider_Registry {
+
+	/**
+	 * An in-memory storage container that holds all registered providers
+	 * for use within the API.
+	 *
+	 * Keyed by the respective provider's unique provider ID:
+	 *
+	 *      @type string $provider_id => @type WP_Webfonts_Provider Provider instance.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @var WP_Webfonts_Provider[]
+	 */
+	private $registered = array();
+
+	/**
+	 * Gets all registered providers.
+	 *
+	 * @since 5.9.0
+	 *
+	 * @return WP_Webfonts_Provider[] All registered providers each keyed by their unique provider ID.
+	 */
+	public function get_all_registered() {
+		return $this->registered;
+	}
+
+	/**
+	 * Initializes the registry.
+	 *
+	 * @since 5.9.0
+	 */
+	public function init() {
+		$this->register_core_providers();
```

Comment:
I'm wondering why the abstraction here?

I'm not seeing this method being used anywhere else, so the contents of it could be in the init method 🤷🏼‍♂️ ?

Reply:
Great question @dingo-d! The original thinking was to (a) encapsulate the knowledge of how-to register the core providers and document it and (b) retain that know-how separately should in the future other unrelated initialization tasks are needed.

What do you think? Does it make the code more or less readable? Do you think it better to move the comments that explain it and the code directly to the `init()` instead of having it as a specific helper task?

Reply:
From the readability perspective, it makes total sense, plus I didn't think far into the future if this would be extended and other init tasks added 😄.

I think it should be left like this. Great explanation 👍🏼